### ISSUE TITLE
Installation with pip

### ISSUE THREAD

**Issue by LucaMantani** [LHCfitNikhef/smefit_release#59]
**Describe the bug**
I tried to install smefit with pip and get the error:

```
╰─ pip install smefit                                                                                                     ─╯
Collecting smefit
  Using cached smefit-0.0.0-py3-none-any.whl.metadata (4.5 kB)
Collecting click<9.0.0,>=8.1.3 (from smefit)
  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Collecting cma<4.0.0,>=3.2.2 (from smefit)
  Using cached cma-3.3.0-py3-none-any.whl (260 kB)
Collecting matplotlib<4.0.0,>=3.5.1 (from smefit)
  Using cached matplotlib-3.8.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (5.8 kB)
Collecting numpy<2.0.0,>=1.23.5 (from smefit)
  Using cached numpy-1.26.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (61 kB)
Collecting pandas<2.0.0,>=1.4.1 (from smefit)
  Using cached pandas-1.5.3-cp310-cp310-macosx_11_0_arm64.whl (10.9 MB)
Collecting pyyaml<6.0.0,>=5.3.0 (from smefit)
  Using cached PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [48 lines of output]
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib3/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/Users/luca/miniforge3/envs/smefit/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/luca/miniforge3/envs/smefit/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/luca/miniforge3/envs/smefit/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 271, in <module>
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command
          super().run_command(command)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 321, in run
          self.find_sources()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 329, in find_sources
          mm.run()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 551, in run
          self.add_defaults()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
          sdist.add_defaults(self)
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/command/sdist.py", line 112, in add_defaults
          super().add_defaults()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
          self._add_defaults_ext()
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
        File "<string>", line 201, in get_source_files
        File "/private/var/folders/21/63l8trwn2dz6sl71hfbbfplw0000gn/T/pip-build-env-06udayxj/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

```
**To Reproduce**
```
mamba create -n smefit && mamba activate smefit
mamba install python=3.10
pip install smefit
```

**Comment by arossia94** [comment]
Hi Luca,
             I ran into the same problem and I could solve it by using the first workaround here: https://stackoverflow.com/a/77491847/23001248.

Hope it works for you too.

Best,

**Comment by giacomomagni** [comment]
Thanks @arossia94 for the solution, so basically there is a problem with the `pyyaml`  dependency.

**Comment by arossia94** [comment]
You're welcome, @giacomomagni . It's a relief to see that is affecting many codes, it's not just smefit.

**Comment by giacomomagni** [comment]
I've released a new version `0.0.1` updating the dependencies and now `pip install smefit` should work again.
As usual to install in dev mode you need `pip install -e .`
